js/storage.js

```diff
@@ -0,0 +1,37 @@
+const store = {
+	currentStore: null,
+
+	init() {
+		//get items from storage
+		//add listeners
+	},
+
+	get(key) {
+		// use async, await here because the below returns a promise
+		// empty key, return all items from storage
+		return browser.storage.local.get();
```

Comment:
Make the function async and it still will return a promise when you do the following:
```
const items = await browser.storage.local.get({key});
return items[0] || null;
```

However that will likely make a nicer API shape for a single item here.